docs/code-reviews/recipes/csharp.md

```diff
@@ -57,10 +57,89 @@ If you are currently using the legacy FxCop analyzers, [migrate from FxCop analy
+The StyleCop analyzer is a NuGet package (StyleCop.Analyzers) that can be installed in any of your projects. It's mainly around code style rules and makes sure the team is following the same rules without having subjective discussions about braces and spaces. Detailed information can be found here: [StyleCop Analyzers for the .NET Compiler Platform](https://github.com/DotNetAnalyzers/StyleCopAnalyzers).
 
 The minimum rules set teams should adopt is the [Managed Recommended Rules](https://msdn.microsoft.com/en-us/library/dd264893.aspx) rule set.
 
+### Improving Code Quality on an Existing Code Base
+
+This describes a method to follow to review and improve the quality of an existing code base. This method is independent from the programming language.
+
+- **Step 1:** Gather state-of-the-art code quality guidelines and rules for the language you are including. (See above)
+- **Step 2:** Group the list of rules you selected in Step 1 (such as the [Managed Recommended Rules](https://msdn.microsoft.com/en-us/library/dd264893.aspx) rule set, mentioned above) into the following code quality categories:
+  - "Static Code Analysis," "Test Coverage," "Observability," "Style and Readability," "Code Clarity and Commenting," "Performance," and "Security"
+  - This will help structure and plan your code review exercise.
+- **Step 3:** Define the types of expected mitigation impact for each rule.
+  - For each rule, think about instances where it can be broken. Go through a mental exercise of fixing these, and decide if the change would likely be breaking/impact existing behavior (Invasive), or if it would likely be non-impactful (Non-invasive).
+  - For example, if refactoring a method could have a big impact on the behavior of the method, and would cause a breaking change where any other code references it, that would be considered "invasive".
+- **Step 4:** Define "review phases" for each rule.
+  - For example, group each rule under labels such as "MVP" and "Completion".
+  - "MVP" should focus on improving the debuggability of the code in production, via adding unit tests, continuous integration, overall reducing the chances that breaking changes can be introduced.
+  - "Completion" should contain the rest. "What else needs to be done to reach completion on the review?"
+
+Once all these steps are completed, you can follow this generic code review plan on the code base you are targeting.
+
+### "Treat Warnings as Errors"
+
+As mentioned above, the [.editorconfig](https://docs.microsoft.com/en-us/visualstudio/ide/editorconfig-code-style-settings-reference?view=vs-2019) allows for configuration and overrides of rules. One approach we reccommend is to set all rules to Error as soon as you begin code quality improvement work.
```

Comment:
nit: recommend 
```suggestion
As mentioned above, the [.editorconfig](https://docs.microsoft.com/en-us/visualstudio/ide/editorconfig-code-style-settings-reference?view=vs-2019) allows for configuration and overrides of rules. One approach we recommend is to set all rules to Error as soon as you begin code quality improvement work.
```

Reply:
@ikivanc - good suggestion, I will split this into two PRs and make the updates suggested above. Thanks!

Reply:
I removed the updates to the csharp.md code review recipes. This PR now focuses only on the addition of the technical debt section.

Reply:
Thanks Shane, I changed the label to `agile`.